docs/serial_driver.md

```diff
@@ -1,129 +1,256 @@
 
-!> The bitbang driver causes connection issues with bitbang WS2812 driver
+* In `halconf.h` add the line `#define PAL_USE_CALLBACKS TRUE`.
 
-Along with the generic options above, you must also turn on the `PAL_USE_CALLBACKS` feature in your halconf.h.
+<hr>
 
-### USART Half-duplex
-Targeting STM32 boards where communication is offloaded to a USART hardware device. The advantage over bitbang is that this provides fast and accurate timings. `SERIAL_PIN_TX` for this driver is the configured USART TX pin. As this Pin is configured in open-drain mode an **external pull-up resistor is needed to keep the line high** (resistor values of 1.5k to 8.2k are known to work). To configure it, add this to your rules.mk:
+## USART Half-duplex
+
+Targeting ARM boards based on ChibiOS, where communication is offloaded to a USART hardware device that supports Half-duplex operation. The advantages over bitbanging are fast, accurate timings and reduced CPU usage. Therefore it is advised to choose this driver or the Full-duplex driver whenever possible.
+
+### Pin configuration
+
+```
+  LEFT                      RIGHT  
++-------+  |           |  +-------+
+|       |  R           R  |       |
+|       |  |   SERIAL  |  |       |
+|    TX |-----------------| TX    |
+|       |       VDD       |       |
+|       |-----------------|       |
+|       |       GND       |       |
+|       |-----------------|       |
++-------+                 +-------+
+```
+
+Only one GPIO pin is needed for the Half-duplex driver, as only one wire is used for receiving and transmitting data. This pin is refereed to as the `SERIAL_USART_TX_PIN` in the configuration. Take care that the pin you chose can act as the TX pin of the USART peripheral. A simple TRS or USB cable provides enough conductors for this driver to work. As the split connection is configured to work in open-drain mode, an **external pull-up resistor is needed to keep the line high**. Resistor values of 1.5kΩ to 8.2kΩ are known to work.
+
+### Setup
+
+To use the Half-duplex driver follow these steps to activate it.
+
+1. Change the `SERIAL_DRIVER` to `usart` in your keyboards `rules.mk` file:
 
 ```make
 SERIAL_DRIVER = usart
 ```
 
-Configure the hardware via your config.h:
+2. Configure the hardware of your keyboard via the `config.h` file:
+
 ```c
-#define SOFT_SERIAL_PIN B6         // USART TX pin
-//#define USART1_REMAP             // Remap USART TX and RX pins on STM32F103 MCUs, see table below.
-#define SELECT_SOFT_SERIAL_SPEED 1 // or 0, 2, 3, 4, 5
-                                   //  0: about 460800 baud
-                                   //  1: about 230400 baud (default)
-                                   //  2: about 115200 baud
-                                   //  3: about 57600 baud
-                                   //  4: about 38400 baud
-                                   //  5: about 19200 baud
-#define SERIAL_USART_DRIVER SD1    // USART driver of TX pin. default: SD1
```

Comment:
There's no longer any mention of setting this define, it should also be made clear that SIO would be `#define SERIAL_USART_DRIVER SIOD2` vs `#define SERIAL_USART_DRIVER SD2` for serial.

Reply:
Perfect ta.